zcutil/fetch-params.py

```diff
@@ -0,0 +1,310 @@
+            logging.error("Unable to lock : %s", file)
+
+    def unlock(file):
+        try:
+            hfile = win32file._get_osfhandle(file.fileno())
+            win32file.UnlockFileEx(hfile, 0, 0xffff0000, __overlapped)
+        except:
+            logging.error("Unable to unlock : %s", file)
+
+elif HOST_OS == 'Linux':
+    import fcntl
+
+    PARAMS_DIR += r"/.zcash-params/"
+    LOCK_SH = fcntl.LOCK_SH
+    LOCK_NB = fcntl.LOCK_NB
+    LOCK_EX = fcntl.LOCK_EX
+    logging.debug("Host: %s", HOST_OS)
+    logging.debug("Parameter path: %s", PARAMS_DIR)
+
+    def lock(file, flags):
+        try:
+            fcntl.flock(file.fileno(), flags)
+        except:
+            logging.error("Unable to lock : %s", file)
+
+    def unlock(file):
+        try:
+            fcntl.flock(file.fileno(), fcntl.LOCK_UN)
+        except:
+            logging.error("Unable to unlock : %s", file)
+
+elif HOST_OS == 'Darwin':
+    import fcntl
+
+    PARAMS_DIR += r"/Library/Application Support/ZcashParams/"
+    LOCK_SH = fcntl.LOCK_SH
+    LOCK_NB = fcntl.LOCK_NB
+    LOCK_EX = fcntl.LOCK_EX
+    logging.debug("Host: %s", HOST_OS)
+    logging.debug("Parameter path: %s", PARAMS_DIR)
+    
+    def lock(file, flags):
+        try:
+            fcntl.flock(file.fileno(), flags)
+        except:
+            logging.error("Unable to lock : %s", file)
+
+    def unlock(file):
+        try:
+            fcntl.flock(file.fileno(), fcntl.LOCK_UN)
+        except:
+            logging.error("Unable to unlock : %s", file)
+else:
+    logging.error(" %s is not currently supported.", HOST_OS)
+
+def is_ipfs_daemon_running():
+    #Due to corner cases of graceful ipfs daemon shutdown via CLI,
+    #we check the ipfs swarm to see if it has a valid network state.
+    #Future IPFS CLI updates should address this with `ipfs status`
+
+    if HOST_OS == 'Windows':
+        p1 = Popen(['ipfs.exe', "swarm", "addrs"], stderr=PIPE, stdout=PIPE)
+    else:
+        p1 = Popen(['ipfs', "swarm","addrs"], stderr=PIPE, stdout=PIPE)
+    stdout, stderr =  p1.communicate()
+
+    if p1.returncode == 1:
+        logging.error("IPFS Daemon is not running. Please start it and try again.")
+        return False
+    else:
+        return True
+
+def use_ipfs(filename):
+    logging.debug("Writing to: %s" % PARAMS_DIR + filename)
+    logging.info("Requesting %s from: /ipfs/%s" % (filename,PARAMS[filename]['ipfs_bhash']))
+    if HOST_OS == 'Windows':
+        p1 = Popen(['ipfs.exe', "get", "--output", PARAMS_DIR + filename, "/ipfs/" + PARAMS[filename]['ipfs_bhash']], stdout=PIPE)
+    else:
+        p1 = Popen(['ipfs', "get", "--output", PARAMS_DIR + filename, "/ipfs/" + PARAMS[filename]['ipfs_bhash']] , stdout=PIPE)
+    buffer = p1.communicate()[0]
+
+    if p1.returncode == 1:
+        logging.error(buffer)
+    
+    verify_ipfs(filename, DOWNLOADING)
+
+def use_https(filename):
+    path = PARAMS_DIR + filename + ".dl"
+    url = PARAMS_URL + filename
+    req = Request(url)
```

Comment:
```suggestion
    req = Request(url, headers={"User-Agent": "Chrome"})
```

i get a `urllib.error.HTTPError: HTTP Error 403: Forbidden` unless i fake the user agent in the request.

Reply:
good catch. ty!